### ISSUE TITLE
Rust downloads some trash and makes build checks flaky

### ISSUE THREAD

**Issue by tavplubix** [ClickHouse/ClickHouse#52314]
https://s3.amazonaws.com/clickhouse-builds/PRs/52276/d05b42ffbf6c12d2ea1d9e06289fd27bc091c326/binary_tidy/build_log.log

```
Jul 18 17:08:16 -- Performing Test SUPPORTS_CFLAG_no_unsafe_buffer_usage - Success
Jul 18 17:08:16 -- compiler C   = /usr/bin/clang-16  --gcc-toolchain=/build/cmake/linux/../../contrib/sysroot/linux-x86_64 -fdiagnostics-color=always -Xclang -fuse-ctor-homing -Wno-enum-constexpr-conversion  -UNDEBUG -gdwarf-aranges -pipe -mssse3 -msse4.1 -msse4.2 -mpclmul -mpopcnt -fasynchronous-unwind-tables -falign-functions=32 -mbranches-within-32B-boundaries  -fdiagnostics-absolute-paths -Weverything -Wpedantic -Wno-zero-length-array -Wno-c++98-compat-pedantic -Wno-c++98-compat -Wno-c++20-compat -Wno-sign-conversion -Wno-implicit-int-conversion -Wno-implicit-int-float-conversion -Wno-ctad-maybe-unsupported -Wno-disabled-macro-expansion -Wno-documentation-unknown-command -Wno-double-promotion -Wno-exit-time-destructors -Wno-float-equal -Wno-global-constructors -Wno-missing-prototypes -Wno-missing-variable-declarations -Wno-padded -Wno-switch-enum -Wno-undefined-func-template -Wno-unused-template -Wno-vla -Wno-weak-template-vtables -Wno-weak-vtables -Wno-thread-safety-negative -Wno-enum-constexpr-conversion -Wno-unsafe-buffer-usage -g -O0 -g -gdwarf-4 
Jul 18 17:08:16 -- compiler CXX = /usr/bin/clang++-16  --gcc-toolchain=/build/cmake/linux/../../contrib/sysroot/linux-x86_64 -std=c++20 -fdiagnostics-color=always -Xclang -fuse-ctor-homing -Wno-enum-constexpr-conversion -fsized-deallocation  -UNDEBUG -gdwarf-aranges -pipe -mssse3 -msse4.1 -msse4.2 -mpclmul -mpopcnt -fasynchronous-unwind-tables -falign-functions=32 -mbranches-within-32B-boundaries -fdiagnostics-absolute-paths -fstrict-vtable-pointers -Wall -Wextra -Weverything -Wpedantic -Wno-zero-length-array -Wno-c++98-compat-pedantic -Wno-c++98-compat -Wno-c++20-compat -Wno-sign-conversion -Wno-implicit-int-conversion -Wno-implicit-int-float-conversion -Wno-ctad-maybe-unsupported -Wno-disabled-macro-expansion -Wno-documentation-unknown-command -Wno-double-promotion -Wno-exit-time-destructors -Wno-float-equal -Wno-global-constructors -Wno-missing-prototypes -Wno-missing-variable-declarations -Wno-padded -Wno-switch-enum -Wno-undefined-func-template -Wno-unused-template -Wno-vla -Wno-weak-template-vtables -Wno-weak-vtables -Wno-thread-safety-negative -Wno-enum-constexpr-conversion -Wno-unsafe-buffer-usage -g -O0 -g -gdwarf-4  -D_LIBCPP_DEBUG=0
Jul 18 17:08:16 -- LINKER_FLAGS =  --gcc-toolchain=/build/cmake/linux/../../contrib/sysroot/linux-x86_64 --ld-path=/usr/bin/ld.lld-16 -rdynamic -Wl,--gdb-index -Wl,--build-id=sha1 -no-pie -Wl,-no-pie 
Jul 18 17:08:16 -- RUST_CFLAGS:  --gcc-toolchain=/build/cmake/linux/../../contrib/sysroot/linux-x86_64 -fdiagnostics-color=always -Xclang -fuse-ctor-homing -Wno-enum-constexpr-conversion  -UNDEBUG -gdwarf-aranges -pipe -mssse3 -msse4.1 -msse4.2 -mpclmul -mpopcnt -fasynchronous-unwind-tables -falign-functions=32 -mbranches-within-32B-boundaries  -fdiagnostics-absolute-paths -Weverything -Wpedantic -Wno-zero-length-array -Wno-c++98-compat-pedantic -Wno-c++98-compat -Wno-c++20-compat -Wno-sign-conversion -Wno-implicit-int-conversion -Wno-implicit-int-float-conversion -Wno-ctad-maybe-unsupported -Wno-disabled-macro-expansion -Wno-documentation-unknown-command -Wno-double-promotion -Wno-exit-time-destructors -Wno-float-equal -Wno-global-constructors -Wno-missing-prototypes -Wno-missing-variable-declarations -Wno-padded -Wno-switch-enum -Wno-undefined-func-template -Wno-unused-template -Wno-vla -Wno-weak-template-vtables -Wno-weak-vtables -Wno-thread-safety-negative -Wno-enum-constexpr-conversion -Wno-unsafe-buffer-usage --sysroot /build/cmake/linux/../../contrib/sysroot/linux-x86_64/x86_64-linux-gnu/libc
Jul 18 17:08:16 -- RUST_CXXFLAGS:  --gcc-toolchain=/build/cmake/linux/../../contrib/sysroot/linux-x86_64 -std=c++20 -fdiagnostics-color=always -Xclang -fuse-ctor-homing -Wno-enum-constexpr-conversion -fsized-deallocation  -UNDEBUG -gdwarf-aranges -pipe -mssse3 -msse4.1 -msse4.2 -mpclmul -mpopcnt -fasynchronous-unwind-tables -falign-functions=32 -mbranches-within-32B-boundaries -fdiagnostics-absolute-paths -fstrict-vtable-pointers -Wall -Wextra -Weverything -Wpedantic -Wno-zero-length-array -Wno-c++98-compat-pedantic -Wno-c++98-compat -Wno-c++20-compat -Wno-sign-conversion -Wno-implicit-int-conversion -Wno-implicit-int-float-conversion -Wno-ctad-maybe-unsupported -Wno-disabled-macro-expansion -Wno-documentation-unknown-command -Wno-double-promotion -Wno-exit-time-destructors -Wno-float-equal -Wno-global-constructors -Wno-missing-prototypes -Wno-missing-variable-declarations -Wno-padded -Wno-switch-enum -Wno-undefined-func-template -Wno-unused-template -Wno-vla -Wno-weak-template-vtables -Wno-weak-vtables -Wno-thread-safety-negative -Wno-enum-constexpr-conversion -Wno-unsafe-buffer-usage -isystem /build/contrib/llvm-project/libcxx/include -nostdinc++ --sysroot /build/cmake/linux/../../contrib/sysroot/linux-x86_64/x86_64-linux-gnu/libc
Jul 18 17:08:16 -- RUSTFLAGS: []
Jul 18 17:08:16 -- RUST_CARGO_BUILD_STD: 
Jul 18 17:08:16 -- Copy BLAKE3 to /build/build_docker/rust/BLAKE3
Jul 18 17:08:16     Updating crates.io index
Jul 18 17:08:16  Downloading crates ...
Jul 18 17:08:16   Downloaded version_check v0.9.4
Jul 18 17:08:16   Downloaded digest v0.9.0
Jul 18 17:08:16   Downloaded generic-array v0.14.6
Jul 18 17:08:16   Downloaded cfg-if v1.0.0
Jul 18 17:08:16   Downloaded arrayvec v0.7.2
Jul 18 17:08:16   Downloaded typenum v1.15.0
Jul 18 17:08:16   Downloaded cc v1.0.73
Jul 18 17:08:16   Downloaded libc v0.2.132
Jul 18 17:08:16   Downloaded constant_time_eq v0.1.5
Jul 18 17:08:16   Downloaded blake3 v1.2.0
Jul 18 17:08:16   Downloaded arrayref v0.3.6
Jul 18 17:08:16 -- Copy skim to /build/build_docker/rust/skim
Jul 18 17:08:16     Updating crates.io index
Jul 18 17:08:17  Downloading crates ...
Jul 18 17:08:17   Downloaded winapi-util v0.1.5
Jul 18 17:08:17 error: failed to download from `https://crates.io/api/v1/crates/dirs-sys-next/0.1.2/download`
Jul 18 17:08:17 
Jul 18 17:08:17 Caused by:
Jul 18 17:08:17   failed to get successful HTTP response from `https://crates.io/api/v1/crates/dirs-sys-next/0.1.2/download` (108.138.64.48), got 421
Jul 18 17:08:17   debug headers:
Jul 18 17:08:17   x-amz-cf-pop: IAD12-P1
Jul 18 17:08:17   x-cache: Error from cloudfront
Jul 18 17:08:17   x-amz-cf-pop: IAD12-P1
Jul 18 17:08:17   x-amz-cf-id: bpeL3EetoD0kWbyEyE8FNcbG0McneFAyNMcX8e2FPLOW1A48-Qjv4A==
Jul 18 17:08:17   body:
Jul 18 17:08:17   <!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN" "http://www.w3.org/TR/html4/loose.dtd">
Jul 18 17:08:17   <HTML><HEAD><META HTTP-EQUIV="Content-Type" CONTENT="text/html; charset=iso-8859-1">
Jul 18 17:08:17   <TITLE>ERROR: The request could not be satisfied</TITLE>
Jul 18 17:08:17   </HEAD><BODY>
Jul 18 17:08:17   <H1>421 ERROR</H1>
Jul 18 17:08:17   <H2>The request could not be satisfied.</H2>
Jul 18 17:08:17   <HR noshade size="1px">
Jul 18 17:08:17   The distribution does not match the certificate for which the HTTPS connection was established with.
Jul 18 17:08:17   We can't connect to the server for this app or website at this …
Jul 18 17:08:17 CMake Error at contrib/corrosion/cmake/CorrosionGenerator.cmake:10 (execute_process):
Jul 18 17:08:17   execute_process failed command indexes:
Jul 18 17:08:17 
Jul 18 17:08:17     1: "Child return code: 101"
Jul 18 17:08:17 
Jul 18 17:08:17 Call Stack (most recent call first):
Jul 18 17:08:17   contrib/corrosion/cmake/CorrosionGenerator.cmake:412 (_cargo_metadata)
Jul 18 17:08:17   contrib/corrosion/cmake/Corrosion.cmake:552 (_generator_add_cargo_targets)
Jul 18 17:08:17   rust/CMakeLists.txt:58 (corrosion_import_crate)
Jul 18 17:08:17   build_docker/rust/skim/CMakeLists.txt:17 (clickhouse_import_crate)
Jul 18 17:08:17 
Jul 18 17:08:17 
Jul 18 17:08:17 -- Configuring incomplete, errors occurred!
```

https://s3.amazonaws.com/clickhouse-builds/PRs/52276/d05b42ffbf6c12d2ea1d9e06289fd27bc091c326/package_msan/build_log.log

```
Jul 18 17:10:56 [8343/11943] cd /build/build_docker/rust/skim && /usr/bin/cmake -E make_directory /build/build_docker/rust/skim/RelWithDebInfo && /usr/bin/cmake -E env CARGO_TARGET_X86_64_UNKNOWN_LINUX_GNU_LINKER=/usr/bin/clang++-16 CC_x86_64-unknown-linux-gnu=/usr/bin/clang-16 HOST_CC=/usr/bin/clang-16 CXX_x86_64-unknown-linux-gnu=/usr/bin/clang++-16 HOST_CXX=/usr/bin/clang++-16 CORROSION_BUILD_DIR=/build/build_docker/rust/skim CARGO_BUILD_RUSTC=/rust/rustup/toolchains/nightly-2023-07-04-x86_64-unknown-linux-gnu/bin/rustc /rust/rustup/toolchains/nightly-2023-07-04-x86_64-unknown-linux-gnu/bin/cargo rustc --target=x86_64-unknown-linux-gnu --package _ch_rust_skim_rust --manifest-path /build/build_docker/rust/skim/Cargo.toml --target-dir /build/build_docker/RelWithDebInfo/cargo/build --profile=release -- -Clink-args=--sysroot=/build/cmake/linux/../../contrib/sysroot/linux-x86_64/x86_64-linux-gnu/libc -Cdefault-linker-libraries=no -Clink-args=--target=x86_64-linux-gnu && /usr/bin/cmake -E copy_if_different /build/build_docker/RelWithDebInfo/cargo/build/x86_64-unknown-linux-gnu/release/lib_ch_rust_skim_rust.a /build/build_docker/rust/skim/RelWithDebInfo
Jul 18 17:10:56     Updating crates.io index
Jul 18 17:10:56  Downloading crates ...
Jul 18 17:10:56 error: failed to download from `https://crates.io/api/v1/crates/allocator-api2/0.2.15/download`
Jul 18 17:10:56 
Jul 18 17:10:56 Caused by:
Jul 18 17:10:56   failed to get successful HTTP response from `https://crates.io/api/v1/crates/allocator-api2/0.2.15/download` (108.138.64.68), got 421
Jul 18 17:10:56   debug headers:
Jul 18 17:10:56   x-amz-cf-pop: IAD12-P1
Jul 18 17:10:56   x-cache: Error from cloudfront
Jul 18 17:10:56   x-amz-cf-pop: IAD12-P1
Jul 18 17:10:56   x-amz-cf-id: 3dpory_1VW7ycll_HCaJYHc6AS-f0A9S6CoFxHNl7TNCwC_-_uw_KQ==
Jul 18 17:10:56   body:
Jul 18 17:10:56   <!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN" "http://www.w3.org/TR/html4/loose.dtd">
Jul 18 17:10:56   <HTML><HEAD><META HTTP-EQUIV="Content-Type" CONTENT="text/html; charset=iso-8859-1">
Jul 18 17:10:56   <TITLE>ERROR: The request could not be satisfied</TITLE>
Jul 18 17:10:56   </HEAD><BODY>
Jul 18 17:10:56   <H1>421 ERROR</H1>
Jul 18 17:10:56   <H2>The request could not be satisfied.</H2>
Jul 18 17:10:56   <HR noshade size="1px">
Jul 18 17:10:56   The distribution does not match the certificate for which the HTTPS connection was established with.
Jul 18 17:10:56   We can't connect to the server for this app or website at this …
Jul 18 17:10:56 [8381/11943] Building CXX object contrib/qpl-cmake/CMakeFiles/_qpl.dir/__/qpl/sources/c_api/compression_operations/compressor.cpp.o
Jul 18 17:10:56 FAILED: rust/skim/CMakeFiles/cargo-build__ch_rust_skim_rust rust/skim/RelWithDebInfo/lib_ch_rust_skim_rust.a /build/build_docker/rust/skim/CMakeFiles/cargo-build__ch_rust_skim_rust /build/build_docker/rust/skim/RelWithDebInfo/lib_ch_rust_skim_rust.a 
Jul 18 17:10:56 cd /build/build_docker/rust/skim && /usr/bin/cmake -E make_directory /build/build_docker/rust/skim/RelWithDebInfo && /usr/bin/cmake -E env CARGO_TARGET_X86_64_UNKNOWN_LINUX_GNU_LINKER=/usr/bin/clang++-16 CC_x86_64-unknown-linux-gnu=/usr/bin/clang-16 HOST_CC=/usr/bin/clang-16 CXX_x86_64-unknown-linux-gnu=/usr/bin/clang++-16 HOST_CXX=/usr/bin/clang++-16 CORROSION_BUILD_DIR=/build/build_docker/rust/skim CARGO_BUILD_RUSTC=/rust/rustup/toolchains/nightly-2023-07-04-x86_64-unknown-linux-gnu/bin/rustc /rust/rustup/toolchains/nightly-2023-07-04-x86_64-unknown-linux-gnu/bin/cargo rustc --target=x86_64-unknown-linux-gnu --package _ch_rust_skim_rust --manifest-path /build/build_docker/rust/skim/Cargo.toml --target-dir /build/build_docker/RelWithDebInfo/cargo/build --profile=release -- -Clink-args=--sysroot=/build/cmake/linux/../../contrib/sysroot/linux-x86_64/x86_64-linux-gnu/libc -Cdefault-linker-libraries=no -Clink-args=--target=x86_64-linux-gnu && /usr/bin/cmake -E copy_if_different /build/build_docker/RelWithDebInfo/cargo/build/x86_64-unknown-linux-gnu/release/lib_ch_rust_skim_rust.a /build/build_docker/rust/skim/RelWithDebInfo
```

**Comment by azat** [comment]
This will be fixed once #49601 will be resolved.

**Comment by rschu1ze** [comment]
Dup of #48575
